mitmproxy/tools/web/app.py

```diff
@@ -290,13 +292,44 @@ def get(self):
 class WebSocketEventBroadcaster(tornado.websocket.WebSocketHandler):
     # raise an error if inherited class doesn't specify its own instance.
     connections: ClassVar[set[WebSocketEventBroadcaster]]
+    application: Application  # reference to the application, in order to be able to update the filters
 
     def open(self, *args, **kwargs):
         self.connections.add(self)
 
     def on_close(self):
         self.connections.discard(self)
 
+    async def on_message(self, message: str):
```

Comment:
Let's move this onto `ClientConnection` class, `WebSocketEventBroadcaster` is just meant to be a generic base class to fan out events to multiple connected clients.